### PR TITLE
dev/core#4213 Make contribution_page.frontend_title required

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#4213 Make frontend_title required

Before
----------------------------------------
frontend_title field optional - leading to inconsistency

After
----------------------------------------
frontend_title field now required - also I added a unique name field

Technical Details
----------------------------------------
https://github.com/civicrm/civicrm-core/pull/25993 was a bit too tricky so this tackles just one of the 3 entities

Comments
----------------------------------------

### CODE REVIEW

**Comment by larssandergreen** [comment]
If we're doing this, maybe we should rename Title (the non-public one), to Name or Page Name, so it's more clear, because it isn't really the title of the page anymore, since it doesn't appear on the page. That's maybe not ideal with the new name field, but I think it would be more clear to users.

**Comment by eileenmcnaughton** [comment]
@larssandergreen good idea - note the name field won't be exposed to users - it's just to have a machine name like other entities

**Comment by larssandergreen** [comment]
@eileenmcnaughton Yes, just confusing for us, but that's bearable.

File: xml/schema/Contribute/ContributionPage.xml [hunk 1]
```diff
@@ -493,18 +518,4 @@
...
-    <type>varchar</type>
-    <length>255</length>
-    <localizable>true</localizable>
-    <default>NULL</default>
-    <comment>Contribution Page Public title</comment>
-    <html>
-      <type>Text</type>
-    </html>
-    <add>5.20</add>
-    <uniqueName>contribution_page_frontend_title</uniqueName>
```
Comment (seamuslee001) [inline xml/schema/Contribute/ContributionPage.xml:508]: @eileenmcnaughton any reason to drop the unique name given frontend_title is used on a few different entities now?

**Comment by eileenmcnaughton** [comment]
@colemanw @seamuslee001 down to 1 fail

- not quite sure if there is a way to rewrite that bit with the exosting localize function

```
Cannot execute UPDATE `civicrm_contribution_page`
SET `frontend_title_en_US` = `title_en_US`
  WHERE `frontend_title_en_US` IS NULL OR `frontend_title_en_US` = '';, UPDATE `civicrm_contribution_page`
SET `frontend_title_af_ZA` = `title_af_ZA`
  WHERE `frontend_title_af_ZA` IS NULL OR `frontend_title_af_ZA` = '';, UPDATE `civicrm_contribution_page`
SET `frontend_title_bg_BG` = `title_bg_BG`
  WHERE `frontend_title_bg_BG` IS NULL OR `frontend_title_bg_BG` = '': DB Error: syntax error
```

**Comment by colemanw** [comment]
You could do a loop like this:
https://github.com/civicrm/civicrm-core/blob/d077a5f64afce475c92ee8e0e1c22da657e2faf0/CRM/Case/xml/configuration.sample/case_sample.mysql.tpl#L34-L42

**Comment by eileenmcnaughton** [comment]
@seamuslee001 @colemanw it passed!!!!

File: CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl [hunk 2]
```diff
@@ -14,3 +14,19 @@ VALUES
     ( {$domainID}, 'https://civicrm.org/help?src=iam',       '{ts escape="sql" skip="true"}Get Help{/ts}',   'Get Help',   NULL, 'AND', @adminHelplastID, '1', NULL, 2 );
 
 UPDATE IGNORE `civicrm_navigation` SET `name` = 'Register Your Site', `label` = '{ts escape="sql" skip="true"}Register Your Site{/ts}' WHERE `name` = 'Register your site';
+
+-- Ensure new name field is not null/unique. Setting to ID is a bit lazy - but sql localisation is painful.
```
Comment (colemanw) [inline CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:18]: @eileenmcnaughton - is this an outdated comment? You seem to have pushed through the pain.

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/sql/5.63.alpha1.mysql.tpl:18]: no - that update is still just setting name to id

**Pull request merged**